src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -126,16 +132,28 @@ async def _handle_unary_unary_rpc(object method_handler,
     )
 
     # Sends response message
-    cdef tuple send_ops = (
-        SendStatusFromServerOperation(
-            tuple(),
-            StatusCode.ok,
-            b'',
-            _EMPTY_FLAGS,
-        ),
-        SendInitialMetadataOperation(None, _EMPTY_FLAGS),
-        SendMessageOperation(response_raw, _EMPTY_FLAGS),
-    )
+    cdef tuple send_ops
+    if servicer_context._metadata_sent:
+        send_ops = (
+            SendStatusFromServerOperation(
```

Comment:
Seems like this block of the conditional has a fair bit in common with the next block. Can we DRY this up a bit?